include/nbl/builtin/hlsl/emulated/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,622 @@
+            if (!FastMath && impl::areBothInfinity(data, rhs.data))
+                return false;
+            if (!FastMath && impl::areBothZero(data, rhs.data))
+                return false;
+
+            const uint64_t lhsSign = ieee754::extractSign(data);
+            const uint64_t rhsSign = ieee754::extractSign(rhs.data);
+
+            // flip bits of negative numbers and flip signs of all numbers
+            uint64_t lhsFlipped = data ^ ((0x7FFFFFFFFFFFFFFFull * lhsSign) | ieee754::traits<float64_t>::signMask);
+            uint64_t rhsFlipped = rhs.data ^ ((0x7FFFFFFFFFFFFFFFull * rhsSign) | ieee754::traits<float64_t>::signMask);
+
+            return lhsFlipped < rhsFlipped;
+        }
+        bool operator>(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            if (!FastMath && (tgmath::isnan<uint64_t>(data) || tgmath::isnan<uint64_t>(rhs.data)))
+                return false;
+            if (!FastMath && impl::areBothInfinity(data, rhs.data))
+                return false;
+            if (!FastMath && impl::areBothZero(data, rhs.data))
+                return false;
+
+            const uint64_t lhsSign = ieee754::extractSign(data);
+            const uint64_t rhsSign = ieee754::extractSign(rhs.data);
+
+            // flip bits of negative numbers and flip signs of all numbers
+            uint64_t lhsFlipped = data ^ ((0x7FFFFFFFFFFFFFFFull * lhsSign) | ieee754::traits<float64_t>::signMask);
+            uint64_t rhsFlipped = rhs.data ^ ((0x7FFFFFFFFFFFFFFFull * rhsSign) | ieee754::traits<float64_t>::signMask);
+
+            return lhsFlipped > rhsFlipped;
+        }
+        bool operator<=(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC 
+        { 
+            if (!FastMath && (tgmath::isnan<uint64_t>(data) || tgmath::isnan<uint64_t>(rhs.data)))
+                return false;
+
+            return !(bit_cast<this_t>(data) > bit_cast<this_t>(rhs.data));
+        }
+        bool operator>=(emulated_float64_t rhs)
+        {
+            if (!FastMath && (tgmath::isnan<uint64_t>(data) || tgmath::isnan<uint64_t>(rhs.data)))
+                return false;
+
+            return !(bit_cast<this_t>(data) < bit_cast<this_t>(rhs.data));
+        }
+
+        //logical operators
+        bool operator&&(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC { return bool(data) && bool(rhs.data); }
+        bool operator||(emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC { return bool(data) || bool(rhs.data); }
+        bool operator!() NBL_CONST_MEMBER_FUNC { return !bool(data); }
+
+        emulated_float64_t flipSign()
+        {
+            return bit_cast<this_t>(data ^ ieee754::traits<float64_t>::signMask);
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE bool supportsFastMath()
+        {
+            return FastMath;
+        }
+
+        enum E_ROUNDING_MODE
+        {
+            FLOAT_ROUND_NEAREST_EVEN,
+            FLOAT_ROUND_TO_ZERO,
+            FLOAT_ROUND_DOWN,
+            FLOAT_ROUND_UP
+        };
```

Comment:
should be defined outside of the templated struct

Reply:
partial specialization of `emulated_float64_t` with additional parameter `E_ROUNDING_MODE` cannot be done due to dxc bug https://github.com/microsoft/DirectXShaderCompiler/issues/5563